models/repo.go

```diff
@@ -1001,6 +999,15 @@ func createDelegateHooks(repoPath string) (err error) {
 		}
 	}
 
+	oldHookPath := filepath.Join(hookDir, "update")
```

Comment:
You should not delete the file and the dir, because user customerize update hook will be stored on `update.d`

Reply:
D'oh. I'm an idiot.